Make userconfirm prompt strings translatable

Userfrom1995 opened the pull request:
The confirmation prompt "Is this ok [Y/n]:" and response characters were hardcoded strings in the userconfirm.hpp template function, making them untranslatable.

Changes:
- Extract translatable strings to libdnf5-cli/utils/userconfirm.cpp
- Add helper functions: get_yes_no_prompt(), get_yes_response(), etc.
- Update libdnf5-cli.pot with new translatable strings

The userconfirm template now calls helper functions that return translated strings from the libdnf5-cli translation domain, allowing translators to localize the prompt and response characters.

Fixes: #1868

Userfrom1995 commented:
> Correct implementation should use nl_langinfo() to retrieve the translated Yes/No expressions. This manual reimplementation is a step in wrong direction.

Thanks for review. I was not aware of nl_langinfo().
I will update the implementation to use it.

evan-goode commented:
Thanks, this does seem to pick up the correct YESEXPR/NOEXPR. However, I still cannot see translations for the "Is this ok [Y/n]: " message, for example, if I add the following to `dnf5/po/es.po`:

```
msgid "Is this ok [Y/n]: "
msgstr "¿Está bien? [S/n]: "

msgid "Is this ok [y/N]: "
msgstr "¿Está bien? [s/N]: "
```

Am I just missing something, were you able to get it working?

Another concern is that the yesexpr/noexpr value and the translated message `[Y/n]` may be or may become out of sync. For example, we don't yet have a Swahili translation, so Swahili-speaking users would likely see the English fallback. Yet according to https://lh.2xlibre.net/values/yesexpr/, the `sw_KE` yesexpr is `^[+1nNyY]`. So I believe with this patch, it would print

```
Is this ok [y/N]:
```

and then if the user types `n`, it would be considered "yes" :upside_down_face:

Userfrom1995 commented:
> > I ran `make libdnf5-cli-pot` to update the template. I updated the .po files using: `cd libdnf5-cli/po && for f in es.po cs.po de.po fr.po; do msgmerge --update "$f" libdnf5-cli.pot; done` I added the translations to the respective files. I committed the changes. I did an RPM build and installed the new binary, and then it worked.
> 
> Thanks! That worked. (Weblate usually takes care of this step for us)
> 
> > Yes, it was a major issue with my previous implementation and I have tried to solve it. Now, the `yesexpr` and `noexpr` from nl_langinfo are used only if a translation is available. If the prompt falls back to English, it strictly uses the "English regex".
> 
> That's a smart and reasonable solution. There is still the risk of a `yesexpr`/`noexpr` changing to become mismatched, but I don't see a way around that problem unless we could somehow dynamically parse the regex and display it in an intuitive way to the user. I am happy with this solution.
> 
We could print the regex pattern, but it would look quite bad, for example:
```Is this ok [^[+1sSyY]/^[-0nN]]:```
Parsing the regex pattern to display it in a user-friendly way would also be too complex for this solution, at least for now. 

> I have one more request: in DNF4 we have
> 
> ```
>             # If the English one letter names don't mix with the translated
>             # letters, allow them too:
>             if u'y' == choice and u'y' not in aui:
>                 choice = yui[0]
>                 break
>             if u'n' == choice and u'n' not in aui:
>                 choice = nui[0]
>                 break
> ```
> 
> could we do the same here? I think we'd just need to keep the old `if (choice == "y" || choice == "Y")` and `if (choice == "n" || choice == "N")` blocks in the loop after the regex tests.

Yes, I just added the conditions. However, there is one interesting thing I noticed on this page: https://lh.2xlibre.net/values/yesexpr/

All the yesexpr regex patterns contain `Yy`, and all the noexpr regex patterns contain `Nn`. So technically, the additional y/n conditions should never be reached unless:

- the regex pattern is missing or corrupted,

- we add a language that does not include `Yy` or `Nn` in its regex pattern, or

- one of the supported languages changes its regex pattern in the future.

So practically speaking, this is more of a safety measure than something that should trigger under normal circumstances.

evan-goode commented:
> Yes, I just added the conditions. However, there is one interesting thing I noticed on this page: https://lh.2xlibre.net/values/yesexpr/
> So practically speaking, this is more of a safety measure than something that should trigger under normal circumstances.

Good point.

I think we're ready to merge, thanks for the patch :)
